services/src/main/java/org/keycloak/protocol/oid4vc/issuance/signing/VCSigningServiceProviderFactory.java

```diff
@@ -0,0 +1,73 @@
+package org.keycloak.protocol.oid4vc.issuance.signing;
+
+import org.keycloak.Config;
+import org.keycloak.common.Profile;
+import org.keycloak.component.ComponentFactory;
+import org.keycloak.component.ComponentModel;
+import org.keycloak.component.ComponentValidationException;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.models.RealmModel;
+import org.keycloak.protocol.oid4vc.model.Format;
+import org.keycloak.provider.ConfigurationValidationHelper;
+import org.keycloak.provider.EnvironmentDependentProviderFactory;
+import org.keycloak.provider.ProviderConfigurationBuilder;
+
+import java.time.Clock;
+
+/**
+ * @author <a href="https://github.com/wistefan">Stefan Wiedemann</a>
+ */
+public interface VCSigningServiceProviderFactory extends ComponentFactory<VerifiableCredentialsSigningService, VerifiableCredentialsSigningService>, EnvironmentDependentProviderFactory {
+    Clock CLOCK = Clock.systemUTC();
```

Comment:
Just a question: how this CLOCK is used?

Reply:
Clock is handed to the concrete signing services, they will need access to time. I could remove them if you prefer and readd once the concrete signing services are added.

Reply:
Thank you for the clarification.

Reply:
fixed [a36c099](https://github.com/keycloak/keycloak/pull/26263/commits/a36c099eb305125846d25f1d73493dc56556573a)

Reply:
Small suggestion: Keycloak uses `org.keycloak.common.util.Time` for accessing the clock, which also allows to provide a global offset.

Reply:
Hi, thanks for mentioning that. Is there also a mechanism to mock time with that class?

Reply:
Yes, +1 for adding something like `Time.currentTimeUTC()` (if using UTC is really needed). We want only one utility for handling "time" in Keycloak, not multiple different utilities :-)